ssg/build_renumber.py

```diff
@@ -269,6 +269,29 @@ def _ensure_by_xccdf_referenced_oval_def_is_defined_in_oval_file(self):
                 # * But not defined in OVAL file
                 rule.remove(check)
 
+    def _ensure_by_xccdf_referenced_oval_no_extra_def_in_oval_file(self):
```

Comment:
If I understand the function correctly. You only want to preserve the OVAL definitions referenced from XCCDF in the OVAL document. I would suggest generating all references for OVAL definitions referenced from XCCDF (aka `references_from_xccdf_to_keep`) and then passing those to the `self.oval_document.keep_referenced_components(references_from_xccdf_to_keep)` method. Then only the OVAL definitions referenced in the XCCDF should be present in the OVAL document, along with all their components. 

```suggestion
    def _ensure_by_xccdf_referenced_oval_no_extra_def_in_oval_file(self):
        # Remove all OVAL checks that are not referenced by XCCDF Rules (checks)
        # or internally via extend-definition

        xccdf_oval_check_refs = [name for name in self._get_list_of_names_of_oval_checks()]
        document_def_keys = list(self.oval_document.definitions.keys())

        references_from_xccdf_to_keep = OVALDefinitionReference()
        for def_id in document_def_keys:
            if def_id in xccdf_oval_check_refs:
                oval_def_refs = self.oval_document.get_all_references_of_definition(def_id)
                references_from_xccdf_to_keep += oval_def_refs

        self.oval_document.keep_referenced_components(references_from_xccdf_to_keep)
```

Reply:
Noice! I'll check it out, thanks.

Reply:
Wonderful. I've simplified the change-set.